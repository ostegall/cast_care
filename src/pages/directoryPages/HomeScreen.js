import { Text, View, Pressable, Image } from 'react-native';
import styles from '../../styles';

export default function HomeScreen({ navigation }) {
  //Screen where user can select which cast issue(s) they are dealing with
  return (
    <View style={{ flex: 1, alignItems: 'center' }}>
      <View style={{ alignItems: 'center', paddingTop: 20, paddingBottom: 30 }}>
        <Text style={styles.title_text}>
          Issue Diagnosis
        </Text>
        <Text style={styles.page_text}>
          What issue(s) can we help you solve?
        </Text>
      </View>
      <View style={{ flexDirection: 'row' }}>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Worse Pain'); /*2x6 (6 rows of 2 buttons) array of buttons with images (see below) 
            that leads to specific cast issue page*/
          }}>
          <Image source={require("../../img/worsepain.png")} /* image provided by client for each issue */ /> 
        </Pressable>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Numbness');
          }}>
          <Image source={require("../../img/numbness.png")} />
        </Pressable>
      </View>
      <View style={{ flexDirection: 'row' }}>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Soiled');
          }}>
          <Image source={require("../../img/soiled.png")} />
        </Pressable>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('No Movement');
          }}>
          <Image source={require("../../img/nomovement.png")} />
        </Pressable>
      </View>
      <View style={{ flexDirection: 'row' }}>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Swelling');
          }}>
          <Image source={require("../../img/swelling.png")} />
        </Pressable>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Burning');
          }}>
          <Image source={require("../../img/burning.png")} />
        </Pressable>
      </View>
      <View style={{ flexDirection: 'row' }}>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Tight');
          }}>
          <Image source={require("../../img/tight.png")} />
        </Pressable>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Loose');
          }}>
          <Image source={require("../../img/loose.png")} />
        </Pressable>
      </View>
      <View style={{ flexDirection: 'row' }}>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Smells');
          }}>
          <Image source={require("../../img/smells.png")} />
        </Pressable>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Rash');
          }}>
          <Image source={require("../../img/rash.png")} />
        </Pressable>
      </View>
      <View style={{ flexDirection: 'row' }}>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Something In Cast');
          }}>
          <Image source={require("../../img/foreignobject.png")} />
        </Pressable>
        <Pressable style={styles.home_button}
          onPress={() => {
            navigation.navigate('Cast Breaking');
          }}>
          <Image source={require("../../img/damaged.png")} />
        </Pressable>
      </View>
    </View>
  );
}