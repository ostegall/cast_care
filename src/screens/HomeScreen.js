import React from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import WorsePain from './WorsePainNumbness';
import Pain from './Pain';
import CastBreaking from './CastBreaking';
import Loose from './Loose';
import Rash from './Rash';
import Smells from './Smells';
import Soiled from './Soiled';
import SomethingInCast from './SomethingInCast';
import Tight from './Tight';

class HomeScreen extends React.Component {
    constructor(props) {
      super(props);
    }

    onPress1 = () => {
      props.navigation.navigate(WorsePain);
    };

      onPress2 = () => {
        props.navigation.navigate(Pain);
    }

      onPress3 = () => {
      props.navigation.navigate(Tight);
    }

      onPress4 = () => {
      props.navigation.navigate(Loose);
    }

      onPress5 = () => {
      props.navigation.navigate(Smells);
    }

      onPress6 = () => {
      props.navigation.navigate(Soiled);
    }

      onPress7 = () => {
      props.navigation.navigate(Rash);
    }

      onPress8 = () => {
      props.navigation.navigate(SomethingInCast);
    }
    
      onPress9 = () => {
      props.navigation.navigate(CastBreaking);
    }

    render () {
      return (
          <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
            <TouchableOpacity onPress={onPress1}>
              <Text>Worsening pain; Numbness; Burning; Severe swelling; No movement?</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={onPress2}>
              <Text>Pain?</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={onPress3}>
              <Text>Feel Too Tight?</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={onPress4}>
              <Text>Feel Too Loose?</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={onPress5}>
              <Text>Smell Bad?</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={onPress6}>
              <Text>Soiled or Wet?</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={onPress7}>
              <Text>Rash or Itching?</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={onPress8}>
              <Text>Something in Cast?</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={onPress9}>
              <Text>Cast Breaking?</Text>
            </TouchableOpacity>
          </View>
      );
    }
}

export default HomeScreen