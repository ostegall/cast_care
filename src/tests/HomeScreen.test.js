import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'

import HomeScreen from '../pages/directoryPages/HomeScreen';

//Makes sure app component renders correctly without crashing
it("renders without crashing", () => {
  shallow(<HomeScreen />);
});

//Snapshot test to ensure UI stays consistent
it('renders correctly', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});