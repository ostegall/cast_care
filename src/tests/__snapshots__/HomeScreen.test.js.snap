// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "paddingBottom": 30,
        "paddingTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 23,
          "fontWeight": "bold",
          "letterSpacing": 0.5,
          "textAlign": "center",
        }
      }
    >
      Issue Diagnosis
    </Text>
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontSize": 17,
          "letterSpacing": 0.5,
          "marginBottom": 5,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      What issue(s) can we help you solve?
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4292c7",
          "borderRadius": 5,
          "elevation": 2,
          "flexDirection": "row",
          "height": 100,
          "justifyContent": "center",
          "minWidth": 88,
          "paddingLeft": 16,
          "paddingRight": 16,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 5,
          "width": 160,
        }
      }
    >
      <Image
        source={1}
      />
    </View>
  </View>
</View>
`;
